scripts/entity/base_entity.gd

```diff
@@ -10,6 +10,8 @@ class_name BaseEntity extends Node
 ## The attack components of the entity.
 @export var attack_component: BaseAttackComponent
 
+@export var terrain_generator: TerrainGenerator
```

Comment:
Feels really weird putting this to an entities job, and like we're asking too much from one source.

Reply:
Yea this absolutely shouldnt be here. Was mostly for testing. I can put it somewhere else if we want temporarily as well.